src/reducers/pipelineRuns.js

```diff
@@ -19,6 +19,14 @@ import { ALL_NAMESPACES } from '../constants';
 
 function byId(state = {}, action) {
   switch (action.type) {
+    case 'PipelineRunCreated':
+    case 'PipelineRunUpdated':
+      const runById = { [action.payload.metadata.uid]: action.payload };
+      return merge({}, state, runById);
+    case 'PipelineRunDeleted':
+      const { ...newState } = state;
```

Comment:
this is typically written the other way around, unless extracting a specific field and lumping everything else together as `...rest`

`const newState = { ...state };` (read 'newState = shallow copy of state')